src/os/FileStore.cc

```diff
@@ -641,10 +642,31 @@ bool parse_attrname(char **name)
 
 void FileStore::collect_metadata(map<string,string> *pm)
 {
+  char partition_path[PATH_MAX];
+  char dev_node[PATH_MAX];
+  int rc = 0;
+  
   (*pm)["filestore_backend"] = backend->get_name();
   ostringstream ss;
   ss << "0x" << std::hex << m_fs_type << std::dec;
   (*pm)["filestore_f_type"] = ss.str();
+
+  rc = get_device_by_uuid(get_fsid(), "PARTUUID", partition_path,
+        dev_node);
+
+  switch (rc) {
+    case -EINVAL:
+      (*pm)["backend_filestore_partition_path"] = "unknown";
+      (*pm)["backend_filestore_dev_node"] = "unknown";
+      break;
+    case -ENODEV:
+      (*pm)["backend_filestore_partition_path"] = string(partition_path);
+      (*pm)["backend_filestore_dev_node"] = "unknown";
+      break;
```

Comment:
You're right, I missed that. I'll add it in the next couple hours.

Joe



On Jun 10, 2015, at 8:56 AM, Haomai Wang <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/os/FileStore.cc<https://github.com/ceph/ceph/pull/4699#discussion_r32119162>:

>    (*pm)["filestore_backend"] = backend->get_name();
>    ostringstream ss;
>    ss << "0x" << std::hex << m_fs_type << std::dec;
>    (*pm)["filestore_f_type"] = ss.str();
> +
> +  rc = get_device_by_uuid(get_fsid(), "PARTUUID", partition_path,
> +        dev_node);
> +
> +  switch (rc) {
> +    case -EINVAL:
> +      (*pm)["backend_filestore_partition_path"] = "unknown";
> +      (*pm)["backend_filestore_dev_node"] = "unknown";
> +      break;
> +    case -ENODEV:
> +      (*pm)["backend_filestore_partition_path"] = string(partition_path);
> +      (*pm)["backend_filestore_dev_node"] = "unknown";
> +      break;


handle -EOPNOTSUPP?

�
Reply to this email directly or view it on GitHub<https://github.com/ceph/ceph/pull/4699/files#r32119162>.